config/config.go

```diff
@@ -165,6 +171,10 @@ var (
 		heights: []uint32{2543000, 0},
 		values:  []int32{14, 0},
 	}
+	NewIDGeneretionStage = HeightDependentInt32{
```

Comment:
I think a more flexible way is to use a few `HeightDependentBool` to control the behavior of each aspect do decouple the "stage" with the behavior in each stage:
`AllowGenerateID1`
`AllowGenerateID2`
`AllowGetID1`